enhancements/single-node/single-node-openshift-with-workers.md

```diff
@@ -0,0 +1,509 @@
+is defaulted to `Workers`. This would be useful to maintain the current behavior
+if the API change is merged before the installer change to set this field, or if
+we want this newer OCP version to be backwards-compatible with older installers.
+
+When the value of `IngressPlacement` is `Workers`, the defaulting behavior of
+`.spec.replicas` and `.spec.nodePlacement` will be the same as it is today:
+`.spec.replicas` will be chosen according to the value of
+`InfrastructureTopology`, namely `1` when `SingleReplica` or `2` when
+`HighlyAvailable`. `.spec.nodePlacement` will always just be:
+
+```yaml
+nodePlacement:
+  nodeSelector:
+    matchLabels:
+      kubernetes.io/os: linux
+      node-role.kubernetes.io/worker: ''
+```
+
+However, if the value of `IngressPlacement` is `ControlPlane`, the defaulting
+behavior will be different: `.spec.replicas` will be chosen instead according
+to the value of `ControlPlaneTopology`; again, `1` when `SingleReplica` or `2`
+when `HighlyAvailable`. `.spec.nodePlacement` will be always just be:
+
+```yaml
+nodePlacement:
+  nodeSelector:
+    matchLabels:
+      kubernetes.io/os: linux
+      node-role.kubernetes.io/master: ''
+```
+
+(Note that the `kubernetes.io/os: linux` label is mentioned just because it's
+the current behavior, it has no importance in this enhancement)
+
+The installer will detect situations in which it's unlikely the user will want
+to set up a load-balancer. For now, those situations include installation of
+single control-plane node cluster deployments on "on-prem" platforms such as
+"none" or "vSphere" (although today single control-plane node clusters are only
+possible on the "none" platform). In those situations, the installer will set
+`IngressPlacement` to be `ControlPlane`. Since there's just a single
+control-plane node, `ControlPlane` topology would be `SingleReplica` and the
+combined effect would be that the `IngressController` will have just a single
+replica and be pinned to the single control-plane node. This will then ensure
+that the `router-default` deployment created by the Cluster Ingress Operator
+will always run on the single control-plane node, and as a result any
+`*.apps.<cluster>.<base>` DNS entries which originally pointed at the single
+control-plane node will remain correct even in the face of newly added worker
+nodes.
+
+In any other situations, the installer will set `IngressPlacement` to
+`Workers`, resulting in the same default behavior as before this enhancement,
+namely that `IngressController` pod replicas are scheduled on worker nodes and
+determined according to the `InfrastructureTopology`.
+
+In the future, when IPI-installed single control-plane node clusters in the
+cloud no longer provision a load-balancer by default, they would also benefit
+from having the installer set the `IngressPlacement` to `ControlPlane`.
+
+### Risks and Mitigations
+
+This should make no noticable difference on "regular" single control-plane node
```

Comment:
What is included in the class of "regular" clusters here?

Reply:
Single control-plane node clusters with no additional worker nodes, will clarify